middleware/session/session.go

```diff
@@ -12,39 +12,78 @@ import (
 	"github.com/valyala/fasthttp"
 )
 
+// Session represents a user session.
 type Session struct {
-	ctx        fiber.Ctx     // fiber context
-	config     *Store        // store configuration
-	data       *data         // key value data
-	byteBuffer *bytes.Buffer // byte buffer for the en- and decode
-	id         string        // session id
-	exp        time.Duration // expiration of this session
-	mu         sync.RWMutex  // Mutex to protect non-data fields
-	fresh      bool          // if new session
+	ctx         fiber.Ctx     // fiber context
+	config      *Store        // store configuration
+	data        *data         // key value data
+	byteBuffer  *bytes.Buffer // byte buffer for encoding/decoding
```

Comment:
can we use bytebufferpool instead?

Reply:
*bytebufferpool.ByteBuffer does not implement io.Reader, so I would have to add a wrapper, however Session never releases the bytes.Buffer and simple calls Reset when it puts it in the Session sync.Pool, so I'm not sure it would be of significant advantage.

Reply:
> *bytebufferpool.ByteBuffer does not implement io.Reader, so I would have to add a wrapper, however Session never releases the bytes.Buffer and simple calls Reset when it puts it in the Session sync.Pool, so I'm not sure it would be of significant advantage.

Yeah but it's a little bit weird since it's a field of a struct. I think separate pool would be cleaner than making it a struct property

Reply:
I removed the code from the Session struct and added a byteBufferPool for encoding and decoding operations. Since not every session will be stored (e.g. new sessions), this change saves time and memory. Tests comparing the old commit with the new one show approximately a 2.17% performance improvement using this method. Thanks for the review!